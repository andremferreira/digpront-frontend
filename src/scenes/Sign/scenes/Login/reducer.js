const login = (state = {}, action) => {
  switch (action.type) {
    case "LOGIN_SUCCESS":
      return action.payload;
    case "LOGIN_ERROR":
    return action.payload;
    default:
      return state;
  }
};

export default login;
