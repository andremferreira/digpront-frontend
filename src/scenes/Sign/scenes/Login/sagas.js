import { put, call } from "redux-saga/effects";
import { startSubmit, stopSubmit, reset } from "redux-form";
import Api from "../../../../api";

export default function* submitLogin(action) {
  try {
    yield put(startSubmit("loginForm"));
    const response = yield call(Api.fetch, "oapi/login", action.payload);
    yield put({ type: "LOGIN_SUCCESS", payload: response });
    yield put(stopSubmit("loginForm"));
    yield put(reset("loginForm"));
  } catch (e) {
    yield put({ type: "LOGIN_ERROR", payload: e.response.data });
    yield put(stopSubmit("loginForm", e.response.data));
  }
}
