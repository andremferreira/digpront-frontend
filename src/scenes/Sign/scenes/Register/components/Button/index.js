import React from "react";

const Button = props => {
  const outline = props.outline !== undefined ? props.outline : "";
  return (
    <button
      type="submit"
      className={`btn ${outline} mr-3 col-2 shadow-sm`}
      disabled={props.submitting || props.invalid}
      onClick={props.onClick}
    >
      {props.name}
    </button>
  );
};

export default Button;
