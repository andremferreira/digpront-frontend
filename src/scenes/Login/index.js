import React from "react";
import Button from "../../components/button";
import CardHeader from "../../components/CardHeader";
import Input from "../../components/input";
import { connect } from "react-redux";
import { Field, reduxForm } from "redux-form";

class Login extends React.Component {

  render() {
    const { handleSubmit } = this.props;
    return (
      <div className="login">
        <form onSubmit={handleSubmit}>
          <CardHeader title="Digital ProntuÃ¡rio">
            <Field
              name="username"
              component={Input}
              //readOnly={readOnly}
              cols="12"
              //normalize={upper}
              //validate={[required]}
            />
            <Field
              name="password"
              component={Input}
              //readOnly={readOnly}
              cols="12"
              //normalize={upper}
              //validate={[required]}
            />
            {/* <Input place="Login" />
          <Input place="Senha" /> */}
            <Button  name="Login" />
          </CardHeader>
        </form>
      </div>
    );
  }
}

Login = reduxForm({
  form: "login"
})(Login);

const mapDispatchToProps = {};
Login = connect(
  null,
  mapDispatchToProps
)(Login);

export default Login;
