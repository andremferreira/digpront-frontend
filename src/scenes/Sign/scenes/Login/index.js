import React from "react";
import Button from "../../../../components/button";
import CardHeader from "../../../../components/cardHeader";
import { connect } from "react-redux";
import { Field, reduxForm } from "redux-form";
import renderInput from "../../../../components/renderInput";
import LinkButton from "../LinkButton";
import { Redirect } from "react-router-dom";
import Alert from "../../../../components/alert";

const required = value => (value ? undefined : "Campo obrigatório!");

class Login extends React.Component {

  componentDidMount() {
    this.props.dispatch({ type: "INIT" });
  }
  
  render() {
    const { handleSubmit, submitValues, submitting, logged, errors } = this.props;
    return (
      <div className="layout-center">
        {logged && <Redirect to="/dashboard" />}

        <form onSubmit={handleSubmit(submitValues)} className="form-signin">
          <CardHeader title="Digital Prontuário" shadow="shadow-sm">
            <Alert errors={errors} />
            <Field
              name="email"
              id="email"
              className="form-control"
              component={renderInput}
              type="text"
              label="Email"
              validate={[required]}
            />
            <Field
              name="password"
              id="password"
              className="form-control"
              component={renderInput}
              type="password"
              label="Senha"
              validate={[required]}
            />
            <Button name="Entrar" submitting={submitting} color="primary" />

            <div className="d-flex justify-content-between mt-2 ">
              <LinkButton color="text-muted" url="/recovery">
                Esqueceu a senha?
              </LinkButton>
              <LinkButton color="text-primary" url="/register">
                Criar Conta
              </LinkButton>
            </div>
          </CardHeader>
        </form>
      </div>
    );
  }
}

Login = reduxForm({ form: "loginForm", initialValues: {} })(Login);

const mapDispatchToProps = dispatch => {
  return {
    submitValues: values => dispatch({ type: "LOGIN_REQUEST", payload: values })
  };
};

const mapStateToProps = state => ({
  logged: state.login.logged,
  errors: state.login.errors
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Login);
