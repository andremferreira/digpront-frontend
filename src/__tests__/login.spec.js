import React from "react";
import { shallow, mount } from "enzyme";
import configureStore from "redux-mock-store";
import Login from "../scenes/Sign/scenes/Login";
import Button from "../components/button";

// const initialState = {
//   login: {
//     logged: true
//   }
// };

// const mockStore = configureStore();
// const store = mockStore(initialState);

// const setup = props => {
//   const component = shallow(<Login store={store} {...props} />);

//   return {
//     component: component,
//     button: component.find("button"),
//     linkButton: component.find(LinkButton),
//     field: component.find("Field"),
//     form: component.find("form")
//   };
// };

const setup = (setupProps = {}) => {
  const store = configureStore()({ login: { logged: true } });
  const wrapper = shallow(<Login store={store} />);

  return {
    store,
    wrapper
  };
};

describe("<Login/>", () => {
  it("renders without crashing", () => {
    const { wrapper } = setup();
    expect(wrapper.shallow()).toMatchSnapshot();
  });
});

it("should render one Button", () => {
  const { wrapper } = setup();
  let button = wrapper.find("button");
  expect(button.at(0).length).toEqual(1);
});

it("should render one Button", () => {
  const { wrapper } = setup();
  let alert = wrapper.find("Alert");
  let field = wrapper.find("Field");

  expect(alert.at().length).toEqual(1);

  expect(field.at(1).length).toEqual(1);
});



// it('onSubmit calls preventDefault to stop form submit', () => {
//   let preventDefault = jest.fn();
//   let { wrapper } = setup();
//   let form = wrapper.shallow().shallow().find('form');
//   console.log(form.debug())
//   form.simulate('submit', { preventDefault });
//   expect(preventDefault).toHaveBeenCalled();
// });

//   it("should render one Form", () => {
//     const { form,component } = setup();
//     //let com = component.toJSON();
//    // let com = renderer.create(component).toJSON()
//     //expect(com).toMatchSnapshot();
//     console.log(form.debug())
//     expect(form.first().length).toEqual(1);
//   });
// });

// it("should render two Field", () => {
//   const { field } = setup();
//   expect(field.at(0).length).toEqual(1);
// });

// it("should submit Button", () => {

//   const { button } = setup(props);
//   console.log(button.debug());
//   button.at(0).simulate("submit", { preventDefault: jest.fn() });
//   expect(actions.onSubmit).toHaveBeenCalledTimes(1);
// });

// it("should submit Button", () => {

//  // const props = {onSubmit:jest.fn()};
//   const { form } = setup();

//   form.first().simulate("submit");
//   expect(
//     form
//       .find("p.error")
//       .first()
//       .text()
//   ).toBe("Please enter your name.");
// });

//component.find('form').simulate('submit', { preventDefault: jest.fn() });
