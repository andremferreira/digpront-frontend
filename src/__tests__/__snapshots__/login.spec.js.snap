// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login/> renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReduxForm
    errors={undefined}
    logged={true}
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
    storeSubscription={
        Subscription {
            "listeners": Object {
              "clear": [Function],
              "get": [Function],
              "notify": [Function],
              "subscribe": [Function],
            },
            "onStateChange": [Function],
            "parentSub": undefined,
            "store": Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            },
            "unsubscribe": [Function],
          }
    }
    submitValues={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "destroyOnUnmount": true,
      "enableReinitialize": false,
      "errors": undefined,
      "forceUnregisterOnUnmount": false,
      "form": "loginForm",
      "getFormState": [Function],
      "initialValues": Object {},
      "keepDirtyOnReinitialize": false,
      "logged": true,
      "persistentSubmitErrors": false,
      "pure": true,
      "shouldAsyncValidate": [Function],
      "shouldError": [Function],
      "shouldValidate": [Function],
      "shouldWarn": [Function],
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
      "submitValues": [Function],
      "touchOnBlur": true,
      "touchOnChange": false,
      "updateUnregisteredFields": false,
    },
    "ref": [Function],
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "destroyOnUnmount": true,
        "enableReinitialize": false,
        "errors": undefined,
        "forceUnregisterOnUnmount": false,
        "form": "loginForm",
        "getFormState": [Function],
        "initialValues": Object {},
        "keepDirtyOnReinitialize": false,
        "logged": true,
        "persistentSubmitErrors": false,
        "pure": true,
        "shouldAsyncValidate": [Function],
        "shouldError": [Function],
        "shouldValidate": [Function],
        "shouldWarn": [Function],
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": [Function],
        },
        "submitValues": [Function],
        "touchOnBlur": true,
        "touchOnChange": false,
        "updateUnregisteredFields": false,
      },
      "ref": [Function],
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
